### PR TITLE
objfw: fetch from the original fossil instance; 1.2 -> 1.2.1

### PR DESCRIPTION
**Pull request by steeleduncan**
The original version of this derivation fetched from a github mirror rather than the original fossil source, so I am updating that. This does not alter the sha of the fetched contents

The second commit bumps the version from 1.2 to 1.2.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ob/objfw/package.nix [hunk 1]
```diff
@@ -3,18 +3,17 @@
...
   pname = "objfw";
   version = "1.2";
 
-  src = fetchFromGitHub {
-    owner = "ObjFW";
-    repo = "ObjFW";
-    rev = "refs/tags/1.2-release";
+  src = fetchfossil {
+    url = "https://objfw.nil.im/home";
+    rev = "7669b5f9d4846fd289d17f2f078c113dc8c92179ce4563256f98850660bb34d4";
```
Comment (Midar) [inline pkgs/by-name/ob/objfw/package.nix]: Maybe you can use a tag here such as `1.2-release`?

File: pkgs/by-name/ob/objfw/package.nix [hunk 2]
```diff
@@ -33,7 +32,7 @@ clangStdenv.mkDerivation (finalAttrs: {
 
   meta = {
     description = "A portable framework for the Objective-C language";
-    homepage = "https://github.com/ObjFW/ObjFW";
+    homepage = "https://objfw.nil.im/home";
```
Comment (Midar) [inline pkgs/by-name/ob/objfw/package.nix:35]: Without `/home`

File: pkgs/by-name/ob/objfw/package.nix (hunk 1 above)
Comment (steeleduncan) [reply pkgs/by-name/ob/objfw/package.nix]: I could well be wrong, but I don't think that works with the fossil fetcher

File: pkgs/by-name/ob/objfw/package.nix [hunk 3]
```diff
@@ -3,19 +3,18 @@
...
+  version = "1.2.1";
 
-  src = fetchFromGitHub {
-    owner = "ObjFW";
-    repo = "ObjFW";
-    rev = "refs/tags/1.2-release";
-    hash = "sha256-nGajiLYwkIDyJujN2zWULkQXKV2A2wzjYl9IoZBU/N4=";
+  src = fetchfossil {
+    url = "https://objfw.nil.im/home";
+    rev = "559577af46b079d5d096971f35d0509c2900cd1d83cbb4a44582192481b18b6f";
```
Comment (khaneliman) [inline pkgs/by-name/ob/objfw/package.nix:16]:
```suggestion
    rev = "${finalAttrs.version}-release";
```
Does this work?

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355278`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>objfw</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>objfw</li>
  </ul>
</details>

File: pkgs/by-name/ob/objfw/package.nix (hunk 3 above)
Comment (steeleduncan) [reply pkgs/by-name/ob/objfw/package.nix]: Thanks, that worked!

I've updated the PR

File: pkgs/by-name/ob/objfw/package.nix (hunk 1 above)
Comment (steeleduncan) [reply pkgs/by-name/ob/objfw/package.nix]: turns out I was wrong, PR is updated now

**Review by khaneliman — APPROVED** [review]

**Pull request merged**